src/vmspawn/vmspawn-util.c

```diff
@@ -44,6 +48,32 @@ int qemu_check_kvm_support(void) {
         return -errno;
 }
 
+int qemu_check_vsock_support(void) {
+        _cleanup_close_ int fd = -EBADF;
+
+        /* We must open the vhost-vsock device here to check whether it exists.
+         *
+         * Just using access will result in a tmpfile being created without the module
+         * actually being loaded.
```

Comment:
this is a bit misleading. maybe:

"Just using access() will just check if the device node exists, but not whether a device driver is behind it (this is a common case since systemd-tmpfiles creates the device node on boot, typically). Hence we open() the path to see if there's actually something behind. If not this should return ENODEV. "

Reply:
Fixed in ef3bb39e80.